src/store/dragRecurrenceModal.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * @copyright Copyright (c) 2021 Richard Steinmetz <richard@steinmetz.cloud>
+ * @author Richard Steinmetz <richard@steinmetz.cloud>
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const state = {
+	show: null,
+	resolve: null,
+	reject: null,
+}
+
+const mutations = {
+	/**
+	 * Show the drag recurrence modal and save the promise callbacks for the modal
+	 *
+	 * @param {object} state The Vuex state
+	 * @param {object} data The destructuring object
+	 * @param {function(boolean): void} data.resolve Call to save the current drag process. Specify thisAndAllFuture via the first argument.
+	 * @param {function(any=): void} data.reject Call to cancel the current drag process. Optionally pass an error.
+	 */
+	showDragRecurrenceModal(state, { resolve, reject }) {
+		state.show = true
+		state.resolve = resolve
+		state.reject = reject
+	},
+
+	/**
+	 * Hide the drag recurrence modal but leave the promise callbacks intact to prevent a deadlock
+	 *
+	 * @param {object} state The Vuex state
+	 */
+	hideDragRecurrenceModal(state) {
+		state.show = false
+	},
+}
+
+const getters = {}
+
+const actions = {
+	/**
+	 * Show the drag recurrence modal and save the promise callbacks for the modal
+	 *
+	 * Automatically hides it after the promise is resolved or rejected (finally)
+	 *
+	 * @param {object} data The destructuring object for Vuex
+	 * @param {Function} data.commit The Vuex commit function
+	 * @return {Promise<boolean>} Will resolve with thisAndAllFuture
+	 */
+	async showDragRecurrenceModal({ commit }) {
+		try {
+			return await new Promise((resolve, reject) => {
```

Comment:
do you need to defer this or why is it wrappen in a Promise?

Reply:
We need to store the resolve and rejects callbacks in the globale vue store. The dragging logic is decoupled from our vue frontend and that's why we (unfortunately) have to use the global store to communicate.

The flow is:
1. Dragging logic requests a drag recurrence modal.
2. Resolve and reject are stored in the store. The modal is shown from the Calendar view.
3. Modal is closed -> resolve/reject is called.
4. Control is returned to the dragging logic via this `return await`.

Reply:
That is the ugly part of this solution I was talking about.